posthog/queries/person_query.py

```diff
@@ -140,9 +140,8 @@ def is_used(self):
         if any(self._uses_person_id(prop) for prop in self._filter.property_groups.flat):
             return True
         for entity in self._filter.entities:
-            if entity.math in COUNT_PER_ACTOR_MATH_FUNCTIONS or any(
-                self._uses_person_id(prop) for prop in entity.property_groups.flat
-            ):
+            is_count_per_user = entity.math in COUNT_PER_ACTOR_MATH_FUNCTIONS and entity.math_group_type_index is None
```

Comment:
Same as in column_optimizer: Are we sure we clean the math_group_type_index in the frontend?

It feels like this is better fit via math, but not sure on the context on this.

Reply:
Update: I reworked this to be simpler. Dropped reporting to Sentry for now.